src/dynamic_load_windows.cpp

```diff
@@ -0,0 +1,78 @@
+#include <opentracing/dynamic_load.h>
+#include <opentracing/version.h>
+
+#include <windows.h> 
+
+namespace opentracing {
+BEGIN_OPENTRACING_ABI_NAMESPACE
+namespace {
+
+ class DynamicLibraryHandleWindows : public DynamicLibraryHandle {
+ public:
+   explicit DynamicLibraryHandleWindows(HINSTANCE handle) : handle_{handle} {}
+
+  ~DynamicLibraryHandleWindows() override { FreeLibrary(handle_); }
+
+ private:
+  HINSTANCE handle_;
+};
+
+}  // namespace
+
+expected<DynamicTracingLibraryHandle>
+DynamicallyLoadTracingLibrary(const char* shared_library,
+                              std::string& error_message) noexcept try {
+  const auto handle = LoadLibrary(shared_library);
+  if (handle == nullptr) {
+    error_message = "An error occurred " + GetLastError();
+    return make_unexpected(dynamic_load_failure_error);
+  }
+
+  std::unique_ptr<DynamicLibraryHandle> dynamic_library_handle{
+      new DynamicLibraryHandleWindows{handle}};
+
+  const auto make_tracer_factory =
+      reinterpret_cast<OpenTracingMakeTracerFactoryType**>(
+          GetProcAddress(handle, "OpenTracingMakeTracerFactory"));
+
+  if (make_tracer_factory == nullptr) {
+    error_message = "An error occurred while looking up for OpenTracingMakeTracerFactory : " + GetLastError();
```

Comment:
This [StackOverflow question](https://stackoverflow.com/questions/1387064/how-to-get-the-error-message-from-the-error-code-returned-by-getlasterror) says that we need to use  `FormatMessage` along with `GetLastError` to get a useful message.

Reply:
Good idea. Change done. an example

```
Failed to load tracer library. An error occurred: The specified module could not be found.
```